app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginScreen.kt

```diff
@@ -1,22 +1,174 @@
+
+            focusManager.clearFocus()
+        }
+    }
+
+    Box(
+        modifier = Modifier.fillMaxSize()
+    ) {
+        if(loginState.isLoading) {
+            Spacer(modifier = Modifier.height(8.dp))
+            CircularProgressIndicator(
+                modifier = Modifier.align(alignment = Alignment.Center)
+            )
+        }
+    }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp),
+    ) {
+        Text(text = "Login:")
+        Spacer(modifier = Modifier.height(8.dp))
+
+        // EMAIL
+        EmailField(
+            email = loginState.email,
+            isError = loginState.isInvalidEmail,
+            onEmailChange = {
+                scope.launch {
+                    viewModel.onEvent(LoginEvent.UpdateEmail(it))
+                }
+            }
+        )
+        if(loginState.isInvalidEmail) {
+            Text(text = "Invalid email", color = Color.Red)
+        }
+        Spacer(modifier = Modifier.height(8.dp))
+
+        // PASSWORD
+        PasswordField(
+            password = loginState.password,
+            isError = loginState.isInvalidPassword,
+            onPasswordChange = {
+                scope.launch {
+                    viewModel.onEvent(LoginEvent.UpdatePassword(it))
+                }
+            },
+            isPasswordVisible = loginState.isPasswordVisible,
+            clickTogglePasswordVisibility = {
+                viewModel.sendEvent(
+                    LoginEvent.TogglePasswordVisibility(loginState.isPasswordVisible)
+                )
+            },
+            imeAction = ImeAction.Done,
+            doneAction = {
+                performLogin()
+            }
+        )
+        if(loginState.isInvalidPassword) {
+            Text(text = "Invalid password", color = Color.Red)
+            Spacer(modifier = Modifier.height(8.dp))
+        }
+
+        // LOGIN BUTTON
+        Spacer(modifier = Modifier.height(16.dp))
+        Button(onClick = {
+                performLogin()
+            },
+            enabled = !loginState.isLoading,
+            modifier = Modifier
+                .align(alignment = Alignment.End)
+        ) {
+            Text(text = "Login")
+            if(loginState.isLoading) {
+                CircularProgressIndicator(
+                    modifier = Modifier
+                        .padding(start = 4.dp)
+                        .size(16.dp)
+                        .align(alignment = CenterVertically)
+                )
+            }
+        }
+        Spacer(modifier = Modifier.height(32.dp))
+
+        // REGISTER BUTTON
+        Text(text = "Not a member? Sign up",
+            color = Color.Cyan,
+            modifier = Modifier
+                .align(alignment = Alignment.CenterHorizontally)
+                .clickable(onClick = {
+                    navigator.clearBackStack("LoginScreenDestination")
```

Comment:
I think with this library, you can also pass `LoginScreenDestination` as an object instead of as a string (not 100% sure here). That would have the benefit that if the name of `LoginScreen` would change, the compiler would notify you here, while with a string it would silently break.

Reply:
OK, i will change to the other way of clearing the backstack, without using hard-coded strings.

😺